digits/scheduler.py

```diff
@@ -175,6 +177,30 @@ def add_job(self, job):
             return False
         else:
             self.jobs.append(job)
+
+            if flask._app_ctx_stack.top is not None:
```

Comment:
Ok @lukeyeager,  I'm at an error that has to do with the fact that the test_add_remove_job creates base class Jobs. When they are created or deleted, a message is sent to rebuild the home page and display the recently created or deleted job. That process queries the method Job.job_type(), which raises an exception in the base class.  So, it's the fact that the test creates an instance that is not meant to be instantiated.

The simplest fix seems to be to subclass a TestJob(Job) that implements job_type().

Does that seem reasonable?